### PR TITLE
Epic 11: Internal Social Media Hub extension (Stories 11.1–11.3)

### PR DESCRIPTION
**Pull request by BaggersIO**
## Summary

Adds the internal, `noindex` Events & Resources hub: for any Event or Resource, one branded image plus
four platform-tailored (five text-block) ready-to-copy posts, editable before copying. Closes out Epic
11 (milestone #18) in one bundled PR, matching the playbook used for Epic 10 (#353) and Epic 13 (#341).

- **Story 11.1** (#447) — `src/lib/promo-post.ts` + `promo-platforms.ts`: pure per-platform post-text
  composition (`resolvePromoFacts`, `composePromoPosts`), with a deterministic character-limit drop plan
  that never drops the outward URL, guest attribution, or an injected image-licence notice.
- **Story 11.2** (#448) — `src/lib/promo-image.ts` + `PromoImage.astro`: resolves one branded image per
  entry (an Event's own logo, a Resource's inherited origin-event logo, or a generated branded-card
  fallback) and renders it, surfacing the CC BY-ND licence credit for the one logo that carries one.
- **Story 11.3** (#449) — the integration surface: `/social/events-resources/**` routes rendering both
  modules together via a new prop-driven `HubEntry.astro`, reachable from `/social/` by one bounded link,
  `noindex` solely via the existing `SocialLayout`. Owns this epic's rendered/a11y verification (e2e +
  axe, light/dark, against an enumerated pre-existing contrast baseline).

All three stories landed as one dev-auto run each, independently reviewed (adversarial + edge-case
passes) and verified before commit. 17 non-blocking review findings across the three stories were
triaged: several patched directly (grammar/docstring fixes, missing unit tests, a type-error fix),
the rest logged to `deferred-work.md` (pre-existing patterns, low-severity latent risks, or
explicitly-named follow-ups the stories themselves defer) or rejected as findings that turned out to
match explicit spec decisions on inspection.

## Test plan

- [x] `npx nx test astro` — 257/257 unit tests pass (11.1: 234, 11.2: +16, 11.3: +7)
- [x] `npx nx build astro` — succeeds, `starlight-links-validator` reports all internal links valid
- [x] `cd astro-docs && npx astro check` — 14 errors, unchanged from the pre-existing baseline (none in new files)
- [x] `npx nx run astro:e2e` — new `social-hub.spec.ts` (12 tests) passes: page structure, 5 distinct
      `aria-label`s, copy-after-edit via live clipboard read-back, discovery links, the FrosCon 3-part
      licence-notice chain, and axe light/dark scans (scoped, against the enumerated baseline, plus a
      zero-violation landing-page scan)
- [x] Mechanical scope checks: `git diff --stat` per story lists exactly the declared new/modified files;
      no AD-8-frozen presentational component (`PlatformSection`, `PostText`, `SocialImage`,
      `SocialLayout`'s core mechanism, `EventCard`, `ResourceCard`, `EventsLayout`, `ResourcesLayout`) was
      touched

### CODE REVIEW

File: astro-docs/src/components/social/HubEntry.astro [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+          <PostText platform={post.platform} label={block.label}><div
+            class="hub-post-editable" contenteditable="true" role="textbox" aria-multiline="true"
+            aria-label={`${post.platform} — ${block.label ?? 'post text'}`} spellcheck="true">{block.text}</div></PostText>
+        ))}
+      </PlatformSection>
+    ))
+  }
+</div>
+
+<style>
```
[the remark this answers is not in the document]
Comment (BaggersIO) [reply astro-docs/src/components/social/HubEntry.astro:110]: Already tracked as a deliberate, accepted limitation rather than a gap — this exact concern surfaced independently during this story's own adversarial review pass and is logged in `thymian-internal`'s `deferred-work.md` (gitignored planning repo, so not visible from this checkout).

Story 11.3's own AC7/Dev Notes call this out explicitly: *"the character counter initialises once... and only ever adds over-limit, so it reports the generated length and never tracks edits. Record it as deferred; do not modify PostText (AD-8)."*

The suggested fix (a hub-local `input` listener recomputing `.char-counter`) was considered and intentionally not taken: it would duplicate `PostText`'s internal counting/formatting logic in a second place, which is exactly the kind of second-owner drift this epic's modules (11.1/11.2/11.3) are built to avoid. The clean fix is a `PostText` enhancement to accept a live-updating mode, but that's a change to the frozen presentational component (AD-8) and belongs to a focused follow-up story, not this bundled PR.

Leaving as deferred — will not block this PR.

**Comment by BaggersIO** [comment]
Needs a fresh start.
